### DISCUSSION TITLE
BUG收集

### DISCUSSION THREAD

**Discussion by Passer1072** [Passer1072/RookieAI_yolov8#26] [General]
遇到了BUG？在这里告诉我！

**Comment by Nichenx** [comment]
请教问题：下了运行环境，能进加载界面，鼠标能动，但好像不能识别，会跳提示，用ai识别了一下，说是我安装的CUDA和PyTorch可能不匹配不是最新版，但一更新反而进不去了，安装回原来的又卡在这一步。
报错提示如下：
当前截图模式：mss
FPS: 58
FPS: 82
FPS: 37
Exception in thread Thread-2 (main_program_loop):
Traceback (most recent call last):
  File "D:\python11\Lib\threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "D:\python11\Lib\threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "D:\下载文件\AI控枪\RookieAI_yolov8-main\RookieAI_YOLOv8_V2.5.0.py", line 2303, in main_program_loop
    results = model.predict(frame, save=False, conf=confidence, half=True, agnostic_nms=True, iou=0.7, classes=[
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\python11\Lib\site-packages\ultralytics\engine\model.py", line 563, in predict
    return self.predictor.predict_cli(source=source) if is_cli else self.predictor(source=source, stream=stream)
                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\python11\Lib\site-packages\ultralytics\engine\predictor.py", line 168, in __call__
    return list(self.stream_inference(source, model, *args, **kwargs))  # merge list of Result into one
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\python11\Lib\site-packages\torch\utils\_contextlib.py", line 35, in generator_context
    response = gen.send(None)
               ^^^^^^^^^^^^^^
  File "D:\python11\Lib\site-packages\ultralytics\engine\predictor.py", line 261, in stream_inference
    self.results = self.postprocess(preds, im, im0s)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\python11\Lib\site-packages\ultralytics\models\yolo\detect\predict.py", line 25, in postprocess
    preds = ops.non_max_suppression(
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\python11\Lib\site-packages\ultralytics\utils\ops.py", line 291, in non_max_suppression
    i = torchvision.ops.nms(boxes, scores, iou_thres)  # NMS
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\python11\Lib\site-packages\torchvision\ops\boxes.py", line 41, in nms
    return torch.ops.torchvision.nms(boxes, scores, iou_threshold)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\python11\Lib\site-packages\torch\_ops.py", line 755, in __call__
    return self._op(*args, **(kwargs or {}))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
NotImplementedError: Could not run 'torchvision::nms' with arguments from the 'CUDA' backend. This could be because the operator doesn't exist for this backend, or was omitted during the selective/custom build process (if using custom build). If you are a Facebook employee using PyTorch on mobile, please visit https://fburl.com/ptmfixes for possible resolutions. 'torchvision::nms' is only available for these backends: [CPU, Meta, QuantizedCPU, BackendSelect, Python, FuncTorchDynamicLayerBackMode, Functionalize, Named, Conjugate, Negative, ZeroTensor, ADInplaceOrView, AutogradOther, AutogradCPU, AutogradCUDA, AutogradXLA, AutogradMPS, AutogradXPU, AutogradHPU, AutogradLazy, AutogradMeta, Tracer, AutocastCPU, AutocastCUDA, FuncTorchBatched, BatchedNestedTensor, FuncTorchVmapMode, Batched, VmapMode, FuncTorchGradWrapper, PythonTLSSnapshot, FuncTorchDynamicLayerFrontMode, PreDispatch, PythonDispatcher].

CPU: registered at C:\actions-runner\_work\vision\vision\pytorch\vision\torchvision\csrc\ops\cpu\nms_kernel.cpp:112 [kernel]
Meta: registered at /dev/null:440 [kernel]
QuantizedCPU: registered at C:\actions-runner\_work\vision\vision\pytorch\vision\torchvision\csrc\ops\quantized\cpu\qnms_kernel.cpp:124 [kernel]
BackendSelect: fallthrough registered at ..\aten\src\ATen\core\BackendSelectFallbackKernel.cpp:3 [backend fallback]
Python: registered at ..\aten\src\ATen\core\PythonFallbackKernel.cpp:154 [backend fallback]
FuncTorchDynamicLayerBackMode: registered at ..\aten\src\ATen\functorch\DynamicLayer.cpp:498 [backend fallback]
Functionalize: registered at ..\aten\src\ATen\FunctionalizeFallbackKernel.cpp:324 [backend fallback]
Named: registered at ..\aten\src\ATen\core\NamedRegistrations.cpp:7 [backend fallback]
Conjugate: registered at ..\aten\src\ATen\ConjugateFallback.cpp:17 [backend fallback]
Negative: registered at ..\aten\src\ATen\native\NegateFallback.cpp:19 [backend fallback]
ZeroTensor: registered at ..\aten\src\ATen\ZeroTensorFallback.cpp:86 [backend fallback]
ADInplaceOrView: fallthrough registered at ..\aten\src\ATen\core\VariableFallbackKernel.cpp:86 [backend fallback]
AutogradOther: registered at ..\aten\src\ATen\core\VariableFallbackKernel.cpp:53 [backend fallback]
AutogradCPU: registered at ..\aten\src\ATen\core\VariableFallbackKernel.cpp:57 [backend fallback]
AutogradCUDA: registered at ..\aten\src\ATen\core\VariableFallbackKernel.cpp:65 [backend fallback]
AutogradXLA: registered at ..\aten\src\ATen\core\VariableFallbackKernel.cpp:69 [backend fallback]
AutogradMPS: registered at ..\aten\src\ATen\core\VariableFallbackKernel.cpp:77 [backend fallback]
AutogradXPU: registered at ..\aten\src\ATen\core\VariableFallbackKernel.cpp:61 [backend fallback]
AutogradHPU: registered at ..\aten\src\ATen\core\VariableFallbackKernel.cpp:90 [backend fallback]
AutogradLazy: registered at ..\aten\src\ATen\core\VariableFallbackKernel.cpp:73 [backend fallback]
AutogradMeta: registered at ..\aten\src\ATen\core\VariableFallbackKernel.cpp:81 [backend fallback]
Tracer: registered at ..\torch\csrc\autograd\TraceTypeManual.cpp:297 [backend fallback]
AutocastCPU: registered at C:\actions-runner\_work\vision\vision\pytorch\vision\torchvision\csrc\ops\autocast\nms_kernel.cpp:34 [kernel]
AutocastCUDA: registered at C:\actions-runner\_work\vision\vision\pytorch\vision\torchvision\csrc\ops\autocast\nms_kernel.cpp:27 [kernel]
FuncTorchBatched: registered at ..\aten\src\ATen\functorch\LegacyBatchingRegistrations.cpp:720 [backend fallback]
BatchedNestedTensor: registered at ..\aten\src\ATen\functorch\LegacyBatchingRegistrations.cpp:746 [backend fallback]
FuncTorchVmapMode: fallthrough registered at ..\aten\src\ATen\functorch\VmapModeRegistrations.cpp:28 [backend fallback]
Batched: registered at ..\aten\src\ATen\LegacyBatchingRegistrations.cpp:1075 [backend fallback]
VmapMode: fallthrough registered at ..\aten\src\ATen\VmapModeRegistrations.cpp:33 [backend fallback]
FuncTorchGradWrapper: registered at ..\aten\src\ATen\functorch\TensorWrapper.cpp:203 [backend fallback]
PythonTLSSnapshot: registered at ..\aten\src\ATen\core\PythonFallbackKernel.cpp:162 [backend fallback]
FuncTorchDynamicLayerFrontMode: registered at ..\aten\src\ATen\functorch\DynamicLayer.cpp:494 [backend fallback]
PreDispatch: registered at ..\aten\src\ATen\core\PythonFallbackKernel.cpp:166 [backend fallback]
PythonDispatcher: registered at ..\aten\src\ATen\core\PythonFallbackKernel.cpp:158 [backend fallback]

**Reply by molanp** [reply]
查询nvdia显卡版本，然后发到这里，我会对版本号对应依赖安装地址进行更新

**Reply by Nichenx** [reply]
nvidia-smi
Sun Aug 11 22:37:54 2024
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 556.12                 Driver Version: 556.12         CUDA Version: 12.5     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                  Driver-Model | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4090      WDDM  |   00000000:01:00.0  On |                  Off |
| 32%   52C    P0            195W /  490W |    8246MiB /  24564MiB |     62%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A      2364    C+G   ...nt.CBS_cw5n1h2txyewy\SearchHost.exe      N/A      |
|    0   N/A  N/A      6020    C+G   ...B\system_tray\lghub_system_tray.exe      N/A      |
|    0   N/A  N/A      7268    C+G   ...ekyb3d8bbwe\PhoneExperienceHost.exe      N/A      |
|    0   N/A  N/A      8156    C+G   C:\Windows\explorer.exe                     N/A      |
|    0   N/A  N/A      8912    C+G   ...oration\NvContainer\nvcontainer.exe      N/A      |
|    0   N/A  N/A      9236    C+G   ...2txyewy\StartMenuExperienceHost.exe      N/A      |
|    0   N/A  N/A     12284    C+G   ...CBS_cw5n1h2txyewy\TextInputHost.exe      N/A      |
|    0   N/A  N/A     13788    C+G   ...5n1h2txyewy\ShellExperienceHost.exe      N/A      |
|    0   N/A  N/A     14896    C+G   ...US\ArmouryDevice\asus_framework.exe      N/A      |
|    0   N/A  N/A     15484    C+G   ...GeForce Experience\NVIDIA Share.exe      N/A      |
|    0   N/A  N/A     15716    C+G   ...GeForce Experience\NVIDIA Share.exe      N/A      |
|    0   N/A  N/A     17996    C+G   ...on\wallpaper_engine\wallpaper64.exe      N/A      |
|    0   N/A  N/A     19792    C+G   ...siveControlPanel\SystemSettings.exe      N/A      |
|    0   N/A  N/A     19800    C+G   ...\cef\cef.win7x64\steamwebhelper.exe      N/A      |
|    0   N/A  N/A     20008    C+G   ...\cef\cef.win7x64\steamwebhelper.exe      N/A      |
|    0   N/A  N/A     20756    C+G   ...m Files\Mozilla Firefox\firefox.exe      N/A      |
|    0   N/A  N/A     21900    C+G   ...m Files\Mozilla Firefox\firefox.exe      N/A      |
|    0   N/A  N/A     23948    C+G   ...\TslGame\Binaries\Win64\TslGame.exe      N/A      |
|    0   N/A  N/A     27936    C+G   ...1.0_x64__8wekyb3d8bbwe\Video.UI.exe      N/A      |
|    0   N/A  N/A     38180    C+G   ...__8wekyb3d8bbwe\WindowsTerminal.exe      N/A      |
+-----------------------------------------------------------------------------------------+

**Reply by Nichenx** [reply]
好的谢谢哥！

**Reply by Nichenx** [reply]
>>> import torch
>>>
>>> # 检查CUDA是否可用
>>> if torch.cuda.is_available():
...     print("CUDA is available.")
...     print("CUDA version:", torch.version.cuda)
...     # 注意：这给出了PyTorch编译时链接的CUDA版本，可能与你系统上安装的CUDA Toolkit版本不同
... else:
...     print("CUDA is not available.")
...
CUDA is available.
CUDA version: 12.1
不知道上面的cuda为什么好像显示是12.5，我又用python3.11查了一下是12.1

**Reply by molanp** [reply]
镜像源未找到对应的cuda版本，请使用以下命令安装(可能需要良好网络环境)
```shell
pip install torch==2.4.0 torchvision==0.19.0 torchaudio==2.4.0 --index-url https://download.pytorch.org/whl/cu124
```

**Reply by Nichenx** [reply]
<img width="837" alt="屏幕截图0813" src="https://github.com/user-attachments/assets/e2aad45d-8c6f-47ea-8661-286c57a5dd12">
啥情况呀，好像变成进不去了。问了ai说更新cuda到12.4，我更新了还是进不去，跳这个提示

**Reply by Nichenx** [reply]
我发现老版本的不会自己卸载，是不是和这个有关，我明天再研究研究

**Reply by Nichenx** [reply]
卸载了四个12.1，剩一个12.4再试试，还是不行诶

**Reply by molanp** [reply]
> 卸载了四个12.1，剩一个12.4再试试，还是不行诶

装一下nvdia force的驱动

**Reply by Nichenx** [reply]
请问是重新装一下nvdia gefore game ready驱动吗？我进去看了一下，我这个都没更新，能直接更新560.81版本的驱动

**Reply by molanp** [reply]
> 请问是重新装一下nvdia gefore game ready驱动吗？我进去看了一下，我这个都没更新，能直接更新560.81版本的驱动

更新

**Reply by Nichenx** [reply]
感谢，反馈一下，昨天安装成功环境能用啦，上次更新了驱动，还是不行。后来找了个大佬分享给他看看，他自己成功运行了，我把我的全部卸载照他py10的一步一步去弄，最后成功了！

**Reply by molanp** [reply]
新版已解决依赖版本问题，运行以下命令来更新依赖以获取更好的性能

```shell
pip install -r requirements.txt -i https://pypi.doubanio.com/simple/
pip install --upgrade torch torchvision torchaudio -f https://mirror.sjtu.edu.cn/pytorch-wheels/torch_stable.html --no-index
```

**Reply by Nichenx** [reply]
感谢提醒，已更新，想用onnx发现报错：获取失败： 'content'
加载模型文件: D:/下载文件/AI控枪/moxing/HG_pubg_320.onnx
半精度推理 设置为: False
WARNING  Unable to automatically guess model task, assuming 'task=detect'. Explicitly define task for your model, i.e. 'task=detect', 'segment', 'classify','pose' or 'obb'.
开始获取版本号......
获取失败： 'content'
Loading settings...
设置加载成功！
update_values function was called（配置已更新）
正在等待加载完成
[INFO]已禁用DXcam
Loading D:\下载文件\AI控枪\moxing\HG_pubg_320.onnx for ONNX Runtime inference...
2024-09-08 12:56:51.2941788 [E:onnxruntime:Default, provider_bridge_ort.cc:1992 onnxruntime::TryGetProviderInfo_CUDA] D:\a\_work\1\s\onnxruntime\core\session\provider_bridge_ort.cc:1637 onnxruntime::ProviderLibrary::Get [ONNXRuntimeError] : 1 : FAIL : LoadLibrary failed with error 126 "" when trying to load "D:\python10\lib\site-packages\onnxruntime\capi\onnxruntime_providers_cuda.dll"

2024-09-08 12:56:51.2994806 [W:onnxruntime:Default, onnxruntime_pybind_state.cc:965 onnxruntime::python::CreateExecutionProviderInstance] Failed to create CUDAExecutionProvider. Require cuDNN 9.* and CUDA 12.*, and the latest MSVC runtime. Please install all dependencies as mentioned in the GPU requirements page (https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements), make sure they're in the PATH, and that your GPU is supported.
Exception in thread Thread-2 (main_program_loop):
Traceback (most recent call last):
  File "D:\python10\lib\threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "D:\python10\lib\threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "D:\下载文件\AI控枪\ai0830\RookieAI_yolov8-main\RookieAI_YOLOv8_V2.5.6.py", line 2575, in main_program_loop
    results = model.predict(frame, save=False, conf=confidence, half=half_precision_model, agnostic_nms=True, iou=0.7, classes=[
  File "D:\python10\lib\site-packages\ultralytics\engine\model.py", line 558, in predict
    self.predictor.setup_model(model=self.model, verbose=is_cli)
  File "D:\python10\lib\site-packages\ultralytics\engine\predictor.py", line 303, in setup_model
    self.model = AutoBackend(
  File "D:\python10\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\python10\lib\site-packages\ultralytics\nn\autobackend.py", line 415, in __init__
    task = metadata["task"]
KeyError: 'task'

**Reply by molanp** [reply]
是否已安装cudnn和cuda驱动

使用此脚本安装
https://github.com/Passer1072/RookieAI_yolov8/blob/main/Tools%2FcuDNN_download_V9.3_12.6.bat

**Reply by Nichenx** [reply]
![图片](https://github.com/user-attachments/assets/555659c4-b649-4d35-b6e4-e5f04253f769)

**Reply by Nichenx** [reply]
看起来似乎应该装好了cuda和cudnn

**Reply by Nichenx** [reply]
![图片](https://github.com/user-attachments/assets/5d65d186-68e6-436b-824d-373289e5cdb1)

**Reply by molanp** [reply]
```shell
pip show torch
pip show torchvision
```
看一下版本号里有没有`cu`

**Reply by Nichenx** [reply]
![图片](https://github.com/user-attachments/assets/d2928d3b-5c02-4200-b488-67cfad97c530)
确实诶，很怪，这个里还是121的cuda

**Reply by molanp** [reply]
目前cu126暂无镜像源，或可使用原版下载url
```
pip install --upgrade torch torchvision -f https://download.pytorch.org/whl/torch_stable.html -no-index
```

**Reply by Nichenx** [reply]
![图片](https://github.com/user-attachments/assets/d7ddbc98-f59a-4ac5-8d30-cf72637f5bb4)
带着-no -index会跳错，我把删掉了进行更新，如图似乎更新成功了，但是onnx还是跳不对：Ultralytics YOLOv8.2.81 🚀 Python-3.10.1 torch-2.4.1+cpu
Exception in thread Thread-2 (main_program_loop):
Traceback (most recent call last):
  File "D:\python10\lib\threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "D:\python10\lib\threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "D:\下载文件\AI控枪\ai0830\RookieAI_yolov8-main\RookieAI_YOLOv8_V2.5.6.py", line 2575, in main_program_loop
    results = model.predict(frame, save=False, conf=confidence, half=half_precision_model, agnostic_nms=True, iou=0.7, classes=[
  File "D:\python10\lib\site-packages\ultralytics\engine\model.py", line 558, in predict
    self.predictor.setup_model(model=self.model, verbose=is_cli)
  File "D:\python10\lib\site-packages\ultralytics\engine\predictor.py", line 305, in setup_model
    device=select_device(self.args.device, verbose=verbose),
  File "D:\python10\lib\site-packages\ultralytics\utils\torch_utils.py", line 174, in select_device
    raise ValueError(
ValueError: Invalid CUDA 'device=0' requested. Use 'device=cpu' or pass valid CUDA device(s) if available, i.e. 'device=0' or 'device=0,1,2,3' for Multi-GPU.

torch.cuda.is_available(): False
torch.cuda.device_count(): 0
os.environ['CUDA_VISIBLE_DEVICES']: None
See https://pytorch.org/get-started/locally/ for up-to-date torch install instructions if no CUDA devices are seen by torch.

**Reply by Nichenx** [reply]
坏消息：鼓捣完这个，不但onnx，原来那个老版本的也打不开了，等周末再来研究研究

**Reply by molanp** [reply]
> ![图片](https://github.com/user-attachments/assets/d7ddbc98-f59a-4ac5-8d30-cf72637f5bb4)
> 带着-no -index会跳错，我把删掉了进行更新，如图似乎更新成功了，但是onnx还是跳不对：Ultralytics YOLOv8.2.81 🚀 Python-3.10.1 torch-2.4.1+cpu
> Exception in thread Thread-2 (main_program_loop):
> Traceback (most recent call last):
>   File "D:\python10\lib\threading.py", line 1009, in _bootstrap_inner
>     self.run()
>   File "D:\python10\lib\threading.py", line 946, in run
>     self._target(*self._args, **self._kwargs)
>   File "D:\下载文件\AI控枪\ai0830\RookieAI_yolov8-main\RookieAI_YOLOv8_V2.5.6.py", line 2575, in main_program_loop
>     results = model.predict(frame, save=False, conf=confidence, half=half_precision_model, agnostic_nms=True, iou=0.7, classes=[
>   File "D:\python10\lib\site-packages\ultralytics\engine\model.py", line 558, in predict
>     self.predictor.setup_model(model=self.model, verbose=is_cli)
>   File "D:\python10\lib\site-packages\ultralytics\engine\predictor.py", line 305, in setup_model
>     device=select_device(self.args.device, verbose=verbose),
>   File "D:\python10\lib\site-packages\ultralytics\utils\torch_utils.py", line 174, in select_device
>     raise ValueError(
> ValueError: Invalid CUDA 'device=0' requested. Use 'device=cpu' or pass valid CUDA device(s) if available, i.e. 'device=0' or 'device=0,1,2,3' for Multi-GPU.
> 
> torch.cuda.is_available(): False
> torch.cuda.device_count(): 0
> os.environ['CUDA_VISIBLE_DEVICES']: None
> See https://pytorch.org/get-started/locally/ for up-to-date torch install instructions if no CUDA devices are seen by torch.

不可以，必须带上那个参数，不然下载的是cpu版本，推荐使用系统代理加速

**Comment by yujuy1686** [comment]
视频预览打不开怎么办啊大佬

<img width="744" height="362" alt="Desktop Screenshot 2026 06 04 - 11 27 06 96 (2)" src="https://github.com/user-attachments/assets/9d9b625d-1aba-4e52-b48c-1808902895b4" />
